deebotozmo/cli.py

```diff
@@ -0,0 +1,591 @@
+
+from deebotozmo.commands import (
+    Charge,
+    CleanCustomArea,
+    CleanPause,
+    CleanResume,
+    CleanSpotArea,
+    CleanStart,
+    PlaySound,
+    SetFanSpeed,
+    SetWaterLevel,
+)
+from deebotozmo.ecovacs_api import EcovacsAPI
+from deebotozmo.events import (
+    BatteryEvent,
+    CleanLogEvent,
+    FanSpeedEvent,
+    MapEvent,
+    RoomsEvent,
+    StatsEvent,
+    StatusEvent,
+    WaterInfoEvent,
+)
+from deebotozmo.models import Vacuum, VacuumState
+from deebotozmo.util import md5
+from deebotozmo.vacuum_bot import VacuumBot
+
+
+def coro(func: Callable) -> Callable:
+    """Wrap around a function to run it in coroutine."""
+
+    @wraps(func)
+    def wrapper(*args: Tuple, **kwargs: Dict) -> Any:
+        loop = asyncio.get_event_loop()
+        return loop.run_until_complete(func(*args, **kwargs))
+
+    return wrapper
+
+
+def config_file() -> str:
+    """Return the path to config file."""
+    if platform.system() == "Windows" and "APPDATA" in os.environ:
+        return os.path.join(str(os.getenv("APPDATA")), "deebotozmo.conf")
+    return os.path.expanduser("~/.config/deebotozmo.conf")
+
+
+def config_file_exists() -> bool:
+    """Check if the config file exists."""
+    return os.path.isfile(config_file())
+
+
+def read_config() -> configparser.ConfigParser:
+    """Read and parse the config file."""
+    parser = configparser.ConfigParser()
+    parser.read(config_file())
+    return parser
+
+
+def read_config_old() -> configparser.SectionProxy:
+    """Read and parser the config file for migration."""
+    parser = configparser.ConfigParser()
+    with open(config_file(), encoding="utf-8") as file:
+        parser.read_file(itertools.chain(["[global]"], file), source=config_file())
+    return parser["global"]
+
+
+def write_config(config: configparser.ConfigParser) -> None:
+    """Create a new config file."""
+    os.makedirs(os.path.dirname(config_file()), exist_ok=True)
+    with open(config_file(), "w", encoding="utf-8") as file:
+        config.write(file)
+
+
+@click.group(chain=True, help="")
+@click.option("--debug/--no-debug", default=False)
+@click.option("--device", default=None, help="Select a device.")
+@click.pass_context
+def cli(
+    ctx: Optional[click.Context] = None,
+    debug: bool = False,
+    device: Optional[str] = None,
+) -> None:
+    """Create a click group for nesting subcommands."""
+    logging.basicConfig(format="%(name)-10s %(levelname)-8s %(message)s")
+    logging.root.setLevel(logging.DEBUG if debug else logging.WARNING)
+
+    # ensure that ctx.obj exists and is a dict (in case `cli()` is called
+    # by means other than the `if` block below)
+    if isinstance(ctx, click.Context):
+        ctx.ensure_object(dict)
+        ctx.obj["DEVICE"] = device
+
+
+# pylint: disable=too-many-arguments
+def _create_config(
+    email: str,
+    password_hash: str,
+    device_id: str,
+    continent_code: str,
+    country_code: str,
+    verify_ssl: str,
+) -> None:
```

Comment:
```suggestion
def _create_config(config: Dict[str, Any]) -> None:
```
please pass the dict instead of each parameter itself